kodilla-rps/src/main/java/InputtingData.java

```diff
@@ -0,0 +1,31 @@
+import java.util.Scanner;
+
+public class InputtingData {
+
+    Scanner scanner = new Scanner(System.in);
+
+    public String introduceText() {
+        String introducedText = scanner.next();
+        return introducedText;
+    }
+
+    public int introductionNumber() throws NumberFormatException {
+        int introductionNumber = 0;
+        try {
+            introductionNumber = Integer.parseInt(introduceText());
+        } catch (NumberFormatException ex) {
+            Messages.showMessageWrongPlayerChoose();
+            introductionNumber();
+        }
+        return introductionNumber;
+    }
+
+    public char validationUserChoiceAfterGame() {
+        char text = scanner.next().toLowerCase().charAt(0);
+        if (text != 'n' & text != 'x') {
+            System.out.println("Dokonano złego wyboru, wprowadź komendę jeszcze");
+            validationUserChoiceAfterGame();
```

Comment:
if user will be patient and provide many times wrong input, you will get StackOverflow. How to avoid it?